frontend/src/security/LoginPage.tsx

```diff
@@ -0,0 +1,75 @@
+import {FormEvent, useState} from "react";
+import axios from "axios";
+import {Link} from "react-router-dom";
+import "./css/LoginPage.css"
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faEye} from "@fortawesome/free-solid-svg-icons";
+import {AppUser} from "./model/AppUser";
+
+const eye = <FontAwesomeIcon icon={faEye}/>;
+
+type Props = {
+    fetchUser: () => void
+    fetchUsername: () => void
+}
+export default function LoginPage(props: Props) {
+    const [passwordShown, setPasswordShown] = useState(false);
+
+    const initialData: AppUser = {
+        "id": "",
+        "username": "",
+        "rawPassword": "",
+        "role": ""
+    }
+    const [appUser, setAppUser] = useState(initialData)
+
+    const togglePassword = () => {
+        setPasswordShown(!passwordShown);
+    };
+
+    const login = () => {
+        axios.get("/api/app-users/login", {
+            auth: {
+                username: appUser.username,
+                password: appUser.rawPassword
+            }
+        })
+            .then(props.fetchUsername)
+            .catch(() => alert("Login failed"));
+    }
+
+    function handleSubmit(event: FormEvent<HTMLFormElement>) {
+        event.preventDefault();
+        login()
+
+    }
+
+    function handleChange(event: any) {
+        setAppUser({
```

Comment:
maybe u can use here CahngeEvent<HTMLInputElement> instead of any? i am not sure